### DISCUSSION TITLE
Got a few errors on building with ninja

### DISCUSSION THREAD

**Discussion by mmdbalkhi** [arch1t3cht/Aegisub#94] [Q&A] [answered]
Hello there! I burned with installing dependencies but could run `meson setup build --buildtype=release` without any error, so I changed root to `build/`, ran `ninja`, and got this [error](https://gist.github.com/mmdbalkhi/66241282026a6882627ffe62f2d77893), it can be cause I Have the Musl?

- Os: Void Linux - musl
- Kernel: `Linux parseh 6.5.7_1 #1 SMP PREEMPT_DYNAMIC Wed Oct 11 08:20:04 UTC 2023 x86_64 GNU/Linux`
- Log: https://gist.github.com/mmdbalkhi/66241282026a6882627ffe62f2d77893

**Comment by 0tkl** [comment]
Use FFMS2 in the fallback subprojects instead of the package provided by your distro. The latest release version of FFMS2 is 3 years old, so it's not compatible with ffmpeg 5.1+.

See https://github.com/arch1t3cht/Aegisub/blob/feature/.github/workflows/ci.yml#L92

**Comment by arch1t3cht** [comment, marked as answer]
The error looks like your ffmpeg is too old for the ffms2 subproject (which requires ffmpeg 6.0 now when building from master).

First of all I'd try a clean rebuild (i.e. recloning the repository from scratch) since some ffms2/ffmpeg related things were changed just a few days ago. In particular meson should now detect if ffmpeg is too old.

If that doesn't work you can try any of
- updating your system ffmpeg to 6.0
- installing ffms2 as a system package (be aware that the currently latest release 2.40 has seeking errors. Though you could still build a ca. one-month-old version from git)
- adding `--force-fallback-for=ffmpeg` to the meson options (this might even happen automatically with a clean rebuild). Note that this will build ffmpeg from scratch with a third-party-wrap, but it can work as a last resort.
- Downgrading the ffms2 subproject to an older version. You'd probably also need to update the `meson.build` file to remove the version requirements for libavcodec/libavformat.

**Reply by mmdbalkhi** [reply]
Well, new errors :smile: https://gist.github.com/mmdbalkhi/30e86beeeb4c102f8989f120d81d4a93

- ffms2 version: ffms2-2.40_1 : (as a system pkg)
- ffmpeg version (built from scratch)
```sh
...
ffmpeg version git-2023-10-17-7218574 Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12.2.0 (GCC)
  configuration: 
  libavutil      58. 27.100 / 58. 27.100
  libavcodec     60. 30.102 / 60. 30.102
  libavformat    60. 15.100 / 60. 15.100
  libavdevice    60.  2.101 / 60.  2.101
  libavfilter     9. 11.100 /  9. 11.100
  libswscale      7.  4.100 /  7.  4.100
  libswresample   4. 11.100 /  4. 11.100
...
```

**Reply by arch1t3cht** [reply]
From ninja's output it doesn't look like meson found your system installation of ffms2 - ninja is still compiling `subprojects/ffms2`. But at least it compiles now.

The new error is due to `RTLD_DEEPBIND` not being defined in `src/vapoursynth_wrap.cpp`. I don't know much about Void Linux and musl but it's possible that that flag is simply not a thing on musl. I can look into that and try to hack together detection for it, but for now as a workaround you can simply remove that flag in `src/vapoursynth_wrap.cpp`, line 41. Or disable VapourSynth support entirely if you don't plan on using it anyway.
